discovery_engine_core/web-api/src/bin/ingestion.rs

```diff
@@ -260,13 +280,36 @@ async fn handle_add_data(
         .map_err(handle_elastic_error)?
         .error_for_status()
         .map_err(handle_elastic_error)?
-        .json::<HashMap<String, serde_json::Value>>()
+        .json::<ElasticIngestionResponse>()
         .await
         .map_err(|err| {
             error!("ReceivingOpError {:#?}", err);
             warp::reject::custom(ReceivingOpError)
         })?;
 
+    if response.errors {
+        let (_, errored_ids): (Vec<_>, Vec<_>) = response
+            .items
+            .into_iter()
+            .map(|hit| {
```

Comment:
:wrench: you can use `filter_map` instead of `map`+`partition`+discarding one parititon